django/contrib/messages/storage/cookie.py

```diff
@@ -75,6 +89,7 @@ class CookieStorage(BaseStorage):
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
         self.signer = signing.get_cookie_signer(salt=self.key_salt)
+        self._not_finished_msg = json.dumps(self.not_finished)
```

Comment:
Minor nitpick, I wonder if we should simply set this to `not_finished_json = '"__messagenotfinished__"'` in the class body, or do we support changes here?

Reply:
I would argue to keep it the way it is, so that, as you mention, we support changing the value (and don't break existing code that is overriding it).
And the performance gains are not significant enough: even if we only call it in the constructor, we build the Storage object for every request, since it's used in the `MessageMiddleware`, but avoiding the `json.dumps` call only brings a few hundreds of nanoseconds of win:
```
In [1]: %timeit json.dumps("__messagesnotfinished__")
302 ns ± 0.167 ns per loop (mean ± std. dev. of 7 runs, 1,000,000 loops each)

In [2]: %timeit '"__messagesnotfinished__"'
4.09 ns ± 0.0443 ns per loop (mean ± std. dev. of 7 runs, 100,000,000 loops each)
```

But I'm open for more discussion, if more people think we should prefer higher performances here.

Reply:
I think `__messagenotfinished__` is an implementation detail and not documented. So we wouldn't have to care much about people trying to change it. That said I am fine with either approach, I'll leave the final decision to you and Mariusz (who has the better idea of whether we try to micro optimize like this).

Reply:
I agree with Florian, we don't support changing `_not_finished_msg`, so avoiding the `json.dumps()` call sounds sensible.

Reply:
Sure, if you both agree 👍 I pushed some edits.